basic_hashtable/b_hashtables.py

```diff
@@ -33,25 +36,62 @@ def hash(string, max):
 # If you are overwriting a value with a different key, print a warning.
 # '''
 def hash_table_insert(hash_table, key, value):
-    pass
-
+    index = hash( key, hash_table.capacity )
+    pair = Pair(key, value)
+
+    if hash_table.storage[index] is not None:
+        #if the bucket is not empty 
+        if hash_table.storage[index].key != key:
+            # checking to see if these keys are different
+            # if there is already something stored at this address print a warning:
+            f"This address already has information stored! are you sure you want to override {hash_table.storage[index].key}?"
+    # if the value des not already exist, add the pair to hash_table:
+    hash_table.storage[index] = pair
```

Comment:
pretty sure this adds the pair no matter what. The warning might give the user false hope... This is the behavior from the spec though.